java/arcs/core/data/CapabilityNew.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+import arcs.core.util.Time
+
+sealed class CapabilityNew {
+    enum class Comparison { LessStrict, Equivalent, Stricter }
+
+    fun isEquivalent(other: CapabilityNew): Boolean {
```

Comment:
Are capabilities totally ordered?

Reply:
sorry, i'm not sure what you mean by this.

Reply:
This is somewhat orthogonal to this PR. Are capabilities a partial order or total order? That is, are all capabilities comparable? Individual capabilities should be. Is it still the case if we consider a cartesian product of capabilities like TTL x Persistence, can we always compare two pairs `(ttl1, persistence1)` and `(ttl2, persistence2)` and say one is less than the other?

Reply:
Not all combinations are comparable. All individual capabilities in a set must be either less strict or equivalent than the other set to be comparable.
For example `disk` is less strict than `memory` and so
(`disk, 1 days`) can't be compared to (`memory, 20 days`)

Reply:
That is a good example. Thanks. This will be relevant when we want to compute capabilities at compile time.

Reply:
yes, and i think i gave a similar example in the ingestion restricting doc.
also, this will be added as part of Capabilities comparison (the class is implemented in ts, kt version is coming next, and comparison still needs to be implemented)